src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/AnyRef.cs

```diff
@@ -21,9 +23,39 @@ internal AnyRef(IntPtr jsHandle, bool ownsHandle) : base(ownsHandle)
         {
             SetHandle(jsHandle);
             AnyRefHandle = GCHandle.Alloc(this, ownsHandle ? GCHandleType.Weak : GCHandleType.Normal);
+            InFlight = null;
+            InFlightCounter = 0;
         }
         internal int Int32Handle => (int)(IntPtr)AnyRefHandle;
 
+        internal void AddInFlight()
+        {
+            lock (this)
+            {
+                InFlightCounter++;
+                if (InFlightCounter == 1 && InFlight == null)
```

Comment:
InFlight should never be able to be != null here, and if it is we should throw
Alternately, let's not use a nullable GCHandle at all because it already has an 'is allocated' property you can query... and we're managing lifetime manually and calling .Free ourselves

Reply:
We want to create AnyRef with unassigned `InFlight`. Unfortunately IntPrt.Zero is not valid GCHandle. So that's why nullable.
I make the null check into assert instead.